src/terminal/parser/stateMachine.cpp

```diff
@@ -2044,3 +2010,26 @@ void StateMachine::_AccumulateTo(const wchar_t wch, size_t& value) noexcept
         value = MAX_PARAMETER_VALUE;
     }
 }
+
+template<typename TLambda>
+bool StateMachine::_SafeExecute(TLambda&& lambda) noexcept
+try
+{
+    return lambda();
+}
+catch (...)
+{
+    LOG_HR(wil::ResultFromCaughtException());
+    return false;
+}
+
+template<typename TLambda>
+bool StateMachine::_SafeExecuteWithLog(const wchar_t wch, TLambda&& lambda) noexcept
+{
+    const bool success = _SafeExecute(std::forward<TLambda>(lambda));
+    if (!success)
+    {
+        TermTelemetry::Instance().LogFailed(wch);
+    }
+    return success;
+}

```

Comment:
Nifty!

Reply:
(maybe something like this belongs in til... but that can be for future consideration)

Reply:
Yeah, that's a good point.